capstone/src/main/java/com/google/sps/servlets/FileUnzipServlet.java

```diff
@@ -0,0 +1,161 @@
+    for (Blob blob1 : blobs.iterateAll()) {
+      System.out.println(blob1.getName());
+    }
+
+    // Retrieve the blob for the apk 
+    Blob blob = storage.get(BlobId.of(bucketName, objectName));
+    System.out.println(blob.getName());
+    
+    return blob;
+
+   
+  }
+
+  public static void analyzeApkFeatures(String nameOfApk, Blob blob, String userId) {
+    
+    byte[] apkBytes = blob.getContent();
+    long dexFileSize = 0, resFileSize = 0, libraryFileSize  = 0, assetsFileSize = 0, resourcesFileSize = 0, miscFileSize = 0, totalApkSize = 0;
+    int filesCount = 0;
+    long timestamp = System.currentTimeMillis();
+
+    try {
+
+      //Declare unzip elements
+      InputStream is = new ByteArrayInputStream(apkBytes);
+      ZipInputStream zis = new ZipInputStream(is);
+      ZipEntry ze = zis.getNextEntry();
+
+      while(ze != null) {
+        String fileName = ze.getName();
+        totalApkSize += ze.getSize();
+        if (fileName.startsWith("res/")) {
+            resFileSize += ze.getSize();
+        } else if (fileName.startsWith("lib/")) {
+            libraryFileSize += ze.getSize();
+        } else if (fileName.startsWith("assets/")) {
+            assetsFileSize += ze.getSize();
+        } else if (fileName.endsWith(".dex")) {
+            dexFileSize += ze.getSize();
+        } else if (fileName.endsWith(".arsc")) {
+            resourcesFileSize += ze.getSize();
+        } else {
+            miscFileSize += ze.getSize();
+        }
+        
+        if (!ze.isDirectory()) {
+          filesCount++;
+        }
+        //close this ZipEntry
+        zis.closeEntry();
+        ze = zis.getNextEntry();
+      }
+
+      //Print the features to the console
+      System.out.println(filesCount);
+      System.out.println("\n" + resFileSize + "\n" + libraryFileSize + "\n" + assetsFileSize + "\n" + dexFileSize + "\n" + resourcesFileSize + "\n" + miscFileSize + "\n");
+      
+      // Initiate the Datastore service for storage of entity created
+      Entity taskEntity = new Entity("UserFileFeature");
+      taskEntity.setProperty("UserId", userId);
+      taskEntity.setProperty("File_Name", nameOfApk);
+      taskEntity.setProperty("Res_File_Size", resFileSize);
+      taskEntity.setProperty("Dex_File_Size", dexFileSize);
+      taskEntity.setProperty("Lib_File_Size", libraryFileSize);
+      taskEntity.setProperty("Asset_File_Size", assetsFileSize);
+      taskEntity.setProperty("Resource_File_Size", resourcesFileSize);
+      taskEntity.setProperty("Misc_File_Size", miscFileSize);
+      taskEntity.setProperty("Total_Apk_size", totalApkSize);
+      taskEntity.setProperty("Files_Count", filesCount);
+      taskEntity.setProperty("Timestamp", timestamp);
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(taskEntity);
+     
+      //close last ZipEntry
+      zis.closeEntry();
+      zis.close();
+
+    } catch (IOException e) {
+      e.printStackTrace();
```

Comment:
Not required, but I would recommend having this method return a boolean and then returning true if it succeeded or return false inside the catch block. Then the servlet can respond with a status 500 error if the unzip fails.